main.sh

```diff
@@ -144,21 +144,28 @@ EOF
 
 # Translate usage string -> getopts arguments, and set $arg_<flag> defaults
 while read -r __b3bp_tmp_line; do
-  if echo "${__b3bp_tmp_line}" |egrep '^-' >/dev/null 2>&1; then
+  if [[ "${__b3bp_tmp_line}" =~ ^- ]]; then
     # fetch single character version of option string
-    __b3bp_tmp_opt="$(echo "${__b3bp_tmp_line}" |awk '{print $1}' |sed -e 's#^-##')"
+    __b3bp_tmp_opt="${__b3bp_tmp_line%% *}"
+    __b3bp_tmp_opt="${__b3bp_tmp_opt:1}"
 
     # fetch long version if present
-    __b3bp_tmp_long_opt="$(echo "${__b3bp_tmp_line}" |awk '/\-\-/ {print $2}' |sed -e 's#^--##')"
-    __b3bp_tmp_long_opt_mangled="$(sed 's#-#_#g' <<< "$__b3bp_tmp_long_opt")"
+    __b3bp_tmp_long_opt=""
+
+    if [[ "${__b3bp_tmp_line}" =~ -- ]]; then
```

Comment:
wait a second. it doesn't matter if this regex is eager or not. all we establish is that a long-opt is there. only the next line actually cuts out the long-opt's name.
I don't think that the behaviour changed to how it was before.
I guess (untested) having the text ` --whatever ` in the description of an option always confused the parser and he might have done something unexpected.
this might warrant a new issue to be opened. not sure how important this is. seems to me the parsing at as it is right now is as good as it gets without blowing the whole code out of proportion.